neqo-http3/src/recv_message.rs

```diff
@@ -467,14 +453,42 @@ impl RecvStream for RecvMessage {
             }
             ResetType::App => {}
         }
+        Ok(())
+    }
+
+    fn stream_type(&self) -> Http3StreamType {
+        Http3StreamType::Http
+    }
+
+    fn http_stream(&mut self) -> Option<&mut dyn HttpRecvStream> {
+        Some(self)
+    }
+}
+
+impl HttpRecvStream for RecvMessage {
+    fn header_unblocked(&mut self, conn: &mut Connection) -> Res<()> {
+        while let Some(p) = self.blocked_push_promise.front() {
+            if let Some(headers) = self
+                .qpack_decoder
+                .borrow_mut()
+                .decode_header_block(&p.header_block, self.stream_id)?
+            {
+                self.push_handler
+                    .as_ref()
+                    .ok_or(Error::HttpFrameUnexpected)?
+                    .borrow_mut()
+                    .new_push_promise(p.push_id, self.stream_id, headers)?;
+                self.blocked_push_promise.pop_front();
+            }
+        }
```

Comment:
I know that this is old code, but...

What happens if a header is unblocked but not all of the push promises can be unblocked?  It looks like this loops forever in that case.  If decode_header_block returns Ok(None) this is an infinite loop.

Reply:
Good catch. Push is not enabled so this is not a problem at the moment, but it needs fixing.